mobile/src/screens/ProfileScreen.tsx

```diff
@@ -120,6 +120,24 @@ const SettingsTab: React.FC = () => {
   const api = useContext(APIContext);
   const auth = useContext(AuthContext);
 
+  const [allowAnalytics , setAllow] = useState(true);
+
+  useEffect(() => {
+    (async () => {
+      const value = await AsyncStorage.getItem('allowAnalytics');
+      if (value === null) {
+        await AsyncStorage.setItem("allowAnalytics", JSON.stringify(allowAnalytics));
+      } else {
+        setAllow(JSON.parse(value) as boolean);
+      }
+    })();
+  }, []);
+
+  const toggleAllowAnalytics = (): void => {
+    setAllow(allowAnalytics => !allowAnalytics);
+    AsyncStorage.setItem("allowAnalytics", JSON.stringify(allowAnalytics));
```

Comment:
This should be `!allowAnalytics`, since we're toggling it.

Reply:
Wouldn't the setAllow line already toggle it?